.github/workflows/build-test.yml

```diff
@@ -143,3 +143,5 @@ jobs:
     permissions:
       contents: write
     uses: ./.github/workflows/publish-release.yml
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
@Mrtenz Will this work? The `NPM_TOKEN` secret is not available outside of the `npm-publish` environment, so I'm wondering whether this will be undefined initially.

Reply:
Oops. I think you're right. We'll need to move this into the `npm-publish` environment instead/as well.

Reply:
Oh, surprisingly, this _does_ work after all. I just tested it: https://github.com/Gudahtt/eslint-config-fork

The secret is carried through despite the top-level job not being in the environment. Maybe it only checks the environment in the job where the secret is directly read :thinking:

Reply:
I tried publishing that test package without these changes, just to confirm this was the cause of the issue, and sure enough it failed: https://github.com/Gudahtt/eslint-config-fork/actions/runs/3493391583/jobs/5848201027

Reply:
> The secret is carried through despite the top-level job not being in the environment. Maybe it only checks the environment in the job where the secret is directly read 🤔

Hmm. Maybe so! Not obvious, but if it's necessary then alright.